src/Equinox.MessageDb/MessageDbClient.fs

```diff
@@ -84,43 +84,28 @@ type MessageDbClient(source: NpgsqlDataSource) =
         version
         (message: IEventData<JsonElement>)
         =
-        cmd.CommandText <- "select 1 from write_message(@Id, @StreamName, @EventType, @Data, @Meta, @ExpectedVersion)"
-
-        cmd.Parameters.AddWithValue("Id", NpgsqlDbType.Text, message.EventId.ToString())
-        |> ignore
-
-        cmd.Parameters.AddWithValue("StreamName", NpgsqlDbType.Text, streamName)
-        |> ignore
-
-        cmd.Parameters.AddWithValue("EventType", NpgsqlDbType.Text, message.EventType)
-        |> ignore
-
-        cmd.Parameters.AddWithValue("Data", NpgsqlDbType.Jsonb, message.Data.GetRawText())
-        |> ignore
-
-        cmd.Parameters.AddWithValue(
-            "Meta",
-            NpgsqlDbType.Jsonb,
-            match message.Meta.ValueKind with
-            | JsonValueKind.Null
-            | JsonValueKind.Undefined -> "null"
-            | _ -> message.Meta.GetRawText()
-        )
-        |> ignore
-
-        cmd.Parameters.AddWithValue("ExpectedVersion", NpgsqlDbType.Bigint, version)
-        |> ignore
+        cmd.CommandText <- "select 1 from write_message(@Id::text, @StreamName, @EventType, @Data, @Meta, @ExpectedVersion)"
+
+        cmd.Parameters.AddWithValue("Id", NpgsqlDbType.Uuid, message.EventId) |> ignore
+        cmd.Parameters.AddWithValue("StreamName", NpgsqlDbType.Text, streamName) |> ignore
+        cmd.Parameters.AddWithValue("EventType", NpgsqlDbType.Text, message.EventType) |> ignore
+        let data, meta = message.Data, message.Meta
+        cmd.Parameters.AddWithValue("Data", NpgsqlDbType.Jsonb, data.GetRawText()) |> ignore
+        let meta =
+            match meta.ValueKind with
+            | JsonValueKind.Null | JsonValueKind.Undefined -> "null"
```

Comment:
does that get mapped to a null value or the string of 4 letters?
Also maybe Undefined should not leak this far down - maybe map to Json NULL when mapping from IEventData to the internal form etc if possible

Reply:
I'm getting the undefined from the JsonElement FsCodec. I'd suggest maybe killing it at the source there?

Repro:
```fsharp
#r "nuget: FsCodec, 3.0.0-rc.7.1"
#r "nuget: FsCodec.SystemTextJson, 3.0.0-rc.7.1"
#r "nuget: TypeShape"

open TypeShape.UnionContract

type Event =
  | Something
  interface IUnionContract

let codec = FsCodec.SystemTextJson.CodecJsonElement.Create<Event>()

let encoded = codec.Encode((), Something)

printfn "%A" (encoded.Meta.ValueKind)
```

this will print "Undefined"

Reply:
Hm `Undefined` is `default(JsonElement)`
In the Cosmos store, I end up doign similar https://github.com/jet/equinox/search?q=undefinedToNull
As long as the code is happy, my main concern is that its wasteful to put the 4 letters `"null"` as a JSON string into a bson field as I'm guessing this code might do (but really I'm just guessing), better to use the specific json value `null`.
maybe https://github.com/jet/equinox/blob/master/src/Equinox.CosmosStore/CosmosStoreSerialization.fs#L24 would be relevant?
Or maybe it's already exactly right...